### PR TITLE
Household controller and API route

### PR DESCRIPTION
**Pull request by LandGod**
Provides the controller file and api file for dealing with households along with one method and one route. Will need to be expanded, but this is a jumping off point. Also add a utils.js to the controllers folder which adds some cross controller utility methods.

### CODE REVIEW

File: controllers/householdController.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        reject(new ValidationError("Household name must be a string. "));
+      }
+      // TODO: The following validation code results in triggering the if statement when it should not and I can't figure out why right now. Not really needed for MVP though.
+      // if (name === '[object Undefined]' || name === undefined || name === null || isNaN(name)) {
+      //   reject(new ValidationError(`"${name}" is not a valid value for Household name.`));
+      // }
+      if (name.length < 3) {
+        reject(new ValidationError("Household name too short"));
+      }
+      if (name.length > 255) {
```
Comment (jennhenn18) [inline controllers/householdController.js]: do we want to shorten this?

File: controllers/householdController.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+      // Validation checks
+      if (typeof (name) !== 'string') {
+        reject(new ValidationError("Household name must be a string. "));
+      }
+      // TODO: The following validation code results in triggering the if statement when it should not and I can't figure out why right now. Not really needed for MVP though.
+      // if (name === '[object Undefined]' || name === undefined || name === null || isNaN(name)) {
+      //   reject(new ValidationError(`"${name}" is not a valid value for Household name.`));
+      // }
+      if (name.length < 3) {
```
Comment (jennhenn18) [inline controllers/householdController.js]: Is this looking at the string length? I think a 3 letter last name is legitimate. Maybe 1 letter or 2?

File: controllers/householdController.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        reject(new ValidationError("MemberId is not a valid mongoose ObjectId"));
+      } 
+
+      // Search households collection for household with that _id in its list of users
+      // Return the _id and name of the household
+      db.Household.find({ 'members': memberId }, '_id name', function (err, household) {
+
+        // Errors cause promise to be rejected and return that error
+        if (err) { reject(err) }
+        else { resolve(household) }
```
Comment (jennhenn18) [inline controllers/householdController.js]: what does resolve do? We will need to return the house name for the dasboard header

File: routes/api/household.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+const mongoose = require("mongoose"); // For working with mongoose Date and ObjectId objects
+const {
+  validateOauthKey,
+  validateObjectId
+} = require("../../controllers/utils"); // Get utils.js for validation functions
+
+// All routes on this page will be appended to this root: "api/household/""
+
+router
+  .route("/")
```
Comment (jennhenn18) [inline routes/api/household.js]: what is triggering this route?

File: controllers/householdController.js (hunk 1 above)
Comment (LandGod) [reply controllers/householdController.js]: Will change

File: controllers/householdController.js (hunk 2 above)
Comment (LandGod) [reply controllers/householdController.js]: Will Change to 2

File: controllers/householdController.js (hunk 1 above)
Comment (LandGod) [reply controllers/householdController.js]: Edit: Will add comment to come back to this later

File: routes/api/household.js (hunk 4 above)
Comment (LandGod) [reply routes/api/household.js]: As per our conversation this resolves to api/household

File: controllers/householdController.js (hunk 3 above)
Comment (LandGod) [reply controllers/householdController.js]: Changed variable name and added explanatory comments to code

**Pull request merged**